spirv_msl.cpp

```diff
@@ -15094,10 +15120,53 @@ void CompilerMSL::analyze_argument_buffers()
 		});
 
 		uint32_t member_index = 0;
+		uint32_t next_arg_buff_index = 0;
 		for (auto &resource : resources)
 		{
 			auto &var = *resource.var;
 			auto &type = get_variable_data_type(var);
+
+			// If needed, synthesize and add padding members.
+			// member_index and next_arg_buff_index are incremented when padding members are added.
+			if (msl_options.pad_argument_buffer_resources)
+			{
+				while (resource.index > next_arg_buff_index)
+				{
+					auto &rez_bind = get_argument_buffer_resource(desc_set, next_arg_buff_index);
+					switch (rez_bind.base_type)
+					{
+					case SPIRType::Void:
+						next_arg_buff_index += add_argument_buffer_padding_buffer_type(buffer_type, member_index, rez_bind.count);
+						break;
+					case SPIRType::Image:
+						next_arg_buff_index += add_argument_buffer_padding_image_type(buffer_type, member_index, rez_bind.count);
+						break;
+					case SPIRType::Sampler:
+						next_arg_buff_index += add_argument_buffer_padding_sampler_type(buffer_type, member_index, rez_bind.count);
+						break;
+					case SPIRType::SampledImage:
+						next_arg_buff_index += add_argument_buffer_padding_image_type(buffer_type, member_index, rez_bind.count);
```

Comment:
Are SampledImages always packed together in IABs? iirc, it's possible to setup a layout in SPIRV-Cross that looks like:
```
struct Foo { texture a [[id(10)]]; sampler a_smplr [[id(400)]]; }
```
so I wonder if we have to emit dummy binds for either the texture side or sampler side depending on how this works.

Reply:
Hmmm...the `MSLResourceBinding` for that would have both texture and sampler. MoltenVK ensures they are packed together in consecutive arguments...but I suppose another client library might have them laid out differently for some reason. We'd have to somehow track during the lookup that "that time I was after the texture, this time I'm after the sampler", etc.

Unless we have need to make it more complicated at the moment, perhaps we could go with this simpler assumption for now, and grow it out if other clients need to support a different approach to adding both textures and samplers from a single `SampledImage` resource.

Reply:
Latest adds separate lookups for the image and sampler components of `SampledImage`.